File: src/main/scala/io/izzel/taboolib/common/event/PlayerSelectLocaleEvent.java
package io.izzel.taboolib.common.event;

import io.izzel.taboolib.module.event.EventNormal;
import org.bukkit.entity.Player;

public class PlayerSelectLocaleEvent extends EventNormal<PlayerSelectLocaleEvent> {

    private final Player player;
    private String locale;

    public PlayerSelectLocaleEvent(Player player, String locale) {
        super(true);
        this.player = player;
        this.locale = locale;
    }

    public Player getPlayer() {
        return this.player;
    }

    public String getLocale() {
        return locale;
    }

    public void setLocale(String locale) {
        this.locale = locale;
    }
}

File: module-lang/src/main/kotlin/taboolib/module/lang/event/PlayerSelectLocaleEvent.kt
package taboolib.module.lang.event

import taboolib.common.platform.ProxyPlayer
import taboolib.common.platform.event.ProxyEvent

/**
 * TabooLib
 * taboolib.module.lang.event.PlayerSelectLocaleEvent
 *
 * @author sky
 * @since 2021/6/18 11:05 下午
 */
class PlayerSelectLocaleEvent(val player: ProxyPlayer, var locale: String) : ProxyEvent() {

    override val allowCancelled: Boolean
        get() = false
}